src/main/java/de/metas/ui/web/request/process/WEBUI_CreateRequest.java

```diff
@@ -75,27 +86,56 @@ protected String doIt() throws Exception
 	private void createRequestFromBPartner(final I_C_BPartner bpartner)
 	{
 		final I_AD_User defaultContact = Services.get(IBPartnerDAO.class).retrieveDefaultContactOrNull(bpartner, I_AD_User.class);
-		
+
 		final List<JSONDocumentChangedEvent> events = new ArrayList<>();
 		events.add(JSONDocumentChangedEvent.replace(I_R_Request.COLUMNNAME_SalesRep_ID, getAD_User_ID()));
 		events.add(JSONDocumentChangedEvent.replace(I_R_Request.COLUMNNAME_C_BPartner_ID, bpartner.getC_BPartner_ID()));
-		if(defaultContact != null)
+		if (defaultContact != null)
 		{
 			events.add(JSONDocumentChangedEvent.replace(I_R_Request.COLUMNNAME_AD_User_ID, defaultContact.getAD_User_ID()));
 		}
-		
+
 		final DocumentPath documentPath = DocumentPath.builder()
 				.setDocumentType(WindowConstants.WINDOWID_R_Request)
 				.setDocumentId(DocumentId.NEW_ID_STRING)
 				.allowNewDocumentId()
 				.build();
-		
+
+		final DocumentId documentId = documentCollection.forDocumentWritable(documentPath, NullDocumentChangesCollector.instance, document -> {
+			document.processValueChanges(events, ReasonSupplier.NONE);
+			return document.getDocumentId();
+		});
+
+		getResult().setRecordToOpen(TableRecordReference.of(I_R_Request.Table_Name, documentId.toInt()), documentPath.getWindowId().toInt(), OpenTarget.SingleDocumentModal);
+	}
+
+	private void createRequestFromShipment(final I_M_InOut shipment)
+	{
+
+		final IBPartnerDAO bPartnerDAO = Services.get(IBPartnerDAO.class);
+		final I_C_BPartner bPartner = bPartnerDAO.getById(shipment.getC_BPartner_ID());
+		final I_AD_User defaultContact = Services.get(IBPartnerDAO.class).retrieveDefaultContactOrNull(bPartner, I_AD_User.class);
+
+		final List<JSONDocumentChangedEvent> events = new ArrayList<>();
```

Comment:
as a convention we use guava's ImmutableList whenever possible. In this case it means to have:
```suggestion
		final ImmutableList.Builder<JSONDocumentChangedEvent> events = ImmutableList.builder();
```

and down at `document.processValueChanges`, you can then have
```java
document.processValueChanges(events.build(), ReasonSupplier.NONE);
```